### ISSUE TITLE
feat: Add verifyBeforeUpdateEmail Functionality To Authentication

### ISSUE THREAD

**Issue by ewwwgiddings** [capawesome-team/capacitor-firebase#739]
### Plugin(s)

- [ ] Analytics
- [ ] App
- [ ] App Check
- [X] Authentication
- [ ] Crashlytics
- [ ] Cloud Firestore
- [ ] Cloud Functions
- [ ] Cloud Messaging
- [ ] Cloud Storage
- [ ] Performance
- [ ] Remote Config

### Current problem

When selecting email enumeration protection it is said that we should not be using `updateEmail()` and instead use `verifyBeforeUpdateEmail()`:
[Firebase Doc Reference](https://firebase.google.com/docs/reference/js/auth.md#verifybeforeupdateemail_09d6f11)

Here is some more information about it:
[Email Enumeration Docs](https://cloud.google.com/identity-platform/docs/admin/email-enumeration-protection)

As said in the above document:
> Users cannot change their email address without first verifying the new address. For example, you can no longer change a user's email address with the [update](https://cloud.google.com/identity-platform/docs/reference/rest/v1/accounts/update) REST API, the setAccountInfo REST API, or the [updateEmail](https://firebase.google.com/docs/reference/js/auth#updateemail) client SDK method on all platforms.
You can use [verifyBeforeUpdateEmail](https://firebase.google.com/docs/reference/js/auth#verifybeforeupdateemail) for Web and Android or [sendEmailVerification(beforeUpdatingEmail:)](https://firebase.google.com/docs/reference/swift/firebaseauth/api/reference/Classes/User#sendemailverificationbeforeupdatingemail:) for iOS instead.

This is a part of the same documentation for the deprecation of `fetchsigninmethodsforemail()` so it has been considered for other methods.

While we can use the JS SDK for this it is not available for the Android and iOS implementations of this plugin as far as I can tell.****

### Preferred solution

Add the ability to use `verifyBeforeUpdateEmail()` and `sendEmailVerification(beforeUpdatingEmail:)` depending on the environment, or have one method that will do it as it differs with iOS under-the-hood.

### Alternative options

I _think_ I can only easily achieve this on web by using JS SDK as noted in some other workarounds.

If there is a way I can achieve this without adding this method to this plugin I would love to hear how.

### Additional context

When attempting to use the `fetchsigninmethodsforemail()` method I Noticed it was deprecated and followed the links to find out I should also not be using `updateEmail()` if I want to turn off email enumeration protection (which imo should be turned on)

### Before submitting

- [X] I have read and followed the [feature request guidelines](https://capawesome.io/contributing/feature-requests/).
- [X] I have attached links to possibly related issues and discussions.

**Comment by robingenz** [comment]
Would you be willing to create a PR?

**Comment by ewwwgiddings** [comment]
Willing to try my hand at it: https://github.com/capawesome-team/capacitor-firebase/pull/741

I haven't had any experience til now working on Capacitor plugins, but I created this PR just now.

I used the `sendEmailVerification` and `updateEmail` methods as a guide and did a bunch of copy/paste/rename from those.

Didn't click any of the buttons on the PR checklist cause I'm not sure how it's tested.

Maybe someone else who has more knowledge on the subject can review, suggest, or edit the PR if it's not far off?
